Add discourse link and update docs (#57)
canonical/alertmanager-k8s-operator: CONTRIBUTING.md

# Contributing to alertmanager-k8s

## Overview

This documents explains the processes and practices recommended for
contributing enhancements or bug fixing to the Alertmanager Charmed Operator.

The intended use case of this operator is to be deployed as part of the
[COS Lite] bundle, although that is not necessary.

## Setup

A typical setup using [snaps](https://snapcraft.io/) can be found in the
[Juju docs](https://juju.is/docs/sdk/dev-setup).

## Developing

- Prior to getting started on a pull request, we first encourage you to open an
  issue explaining the use case or bug.
  This gives other contributors a chance to weigh in early in the process.

> - To author PRs you should know [what is jujuj](https://juju.is/#what-is-juju)

Reviewer: ```suggestion
- To author PRs you should be familiar with [juju](https://juju.is/#what-is-juju)
```

  and [how operators are written](https://juju.is/docs/sdk).
- The best way to get a head start is to join the conversation on our
  [Mattermost channel] or [Discourse].
- All enhancements require review before being merged. Besides the
  code quality and test coverage, the review will also take into
  account the resulting user experience for Juju administrators using
  this charm. To be able to merge you would have to rebase
  onto the `main` branch. We do this to avoid merge commits and to have a
  linear Git history.
- We use [`tox`](https://tox.wiki/en/latest/#) to manage all virtualenvs for
  the development lifecycle.

### Testing
Unit tests are written with the Operator Framework [test harness] and
integration tests are written using [pytest-operator] and [python-libjuju].

> All default tests can be executed by running `tox` without arguments.

Reviewer: This sentence doesn't really convey any information. If you run `tox` without arguments, you will run the default tests, because that is what the default tests are :-)

I would recommend detailing what will run when executing `tox`. Are there any external dependencies that the tox file will not setup? Will it make any unsafe changes to the system running the tests?

Author: Fixed in next commit.

> You can also manually run specific test environment:

Reviewer: ```suggestion
You can also manually run a specific test environment:
```

# Contributing to alertmanager-k8s

## Overview

This documents explains the processes and practices recommended for
contributing enhancements or bug fixing to the Alertmanager Charmed Operator.

The intended use case of this operator is to be deployed as part of the
[COS Lite] bundle, although that is not necessary.

## Setup

A typical setup using [snaps](https://snapcraft.io/) can be found in the
[Juju docs](https://juju.is/docs/sdk/dev-setup).

## Developing

- Prior to getting started on a pull request, we first encourage you to open an
  issue explaining the use case or bug.
  This gives other contributors a chance to weigh in early in the process.
- To author PRs you should be familiar with [juju](https://juju.is/#what-is-juju)
  and [how operators are written](https://juju.is/docs/sdk).
- The best way to get a head start is to join the conversation on our

>   [Mattermost channel] or [Discourse].

Reviewer 2: Link to Mattermost and Discourse?

Author: It's linked! This is a short-form reference syntax. The link are at the bottom.

    --generator-url="http://prometheus.int.example.net"
```

> or using [Alertmanager's HTTP API][Alertmanager API browser],

Reviewer 2: Don't we have an official documentation for this?

Author: It exists, but it is very narrow:
https://www.prometheus.io/docs/alerting/latest/management_api/
